airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/cdc/MongoDbCdcInitializer.java

```diff
@@ -183,9 +213,60 @@ public List<AutoCloseableIterator<AirbyteMessage>> createCdcIterators(
+       * with ALL of the incremental streams configured. This is because if step 1 completes, the initial
+       * load can be considered finished.
+       */
+      return Collections.singletonList(
+          AutoCloseableIterators.concatWithEagerClose(
+              Stream.of(
+                  cdcStreamsStartStatusEmitters,
+                  Collections.singletonList(initialSnapshotIterator),
+                  Collections.singletonList(AutoCloseableIterators.lazyIterator(incrementalIteratorSupplier, null)),
+                  cdcStreamsCompleteStatusEmitters).flatMap(Collection::stream).collect(Collectors.toList()),
+              AirbyteTraceMessageUtility::emitStreamStatusTrace));
+    } else if (initialSnapshotIterators.isEmpty()) {
+      LOGGER.info("Initial load has finished completely - only reading the binlog");
+      /*
+       * In this case, the initial load has completed and only debezium should be run. The iterators
+       * should be run in the following order: 1. Run the debezium iterators with ALL of the incremental
+       * streams configured.
+       */
+      return Collections.singletonList(
+          AutoCloseableIterators.concatWithEagerClose(
+              Stream.of(
+                  cdcStreamsStartStatusEmitters,
+                  Collections.singletonList(AutoCloseableIterators.lazyIterator(incrementalIteratorSupplier, null)),
+                  cdcStreamsCompleteStatusEmitters).flatMap(Collection::stream).collect(Collectors.toList()),
+              AirbyteTraceMessageUtility::emitStreamStatusTrace));
+    } else {
+      LOGGER.info("Initial load is in progress - reading binlog first and then resuming with initial load.");
+      /*
+       * In this case, the initial load has partially completed (WASS case). The iterators should be run
+       * in the following order: 1. Run the debezium iterators with only the incremental streams which
+       * have been fully or partially completed configured. 2. Resume initial load for partially completed
+       * and not started streams. This step will timeout and throw a transient error if run for too long
+       * (> 8hrs by default). 3. Emit a transient error. This is to signal to the platform to restart the
+       * sync to clear the binlog. We cannot simply add the same cdc iterators as their target end
+       * position is fixed to the tip of the binlog at the start of the sync.
+       */
+      return Collections.singletonList(
+          AutoCloseableIterators.concatWithEagerClose(
+              Stream.of(
+                  cdcStreamsStartStatusEmitters,
+                  Collections.singletonList(AutoCloseableIterators.lazyIterator(incrementalIteratorSupplier, null)),
+                  Collections.singletonList(initialSnapshotIterator),
+                  cdcStreamsCompleteStatusEmitters,
+                  List.of(new TransientErrorTraceEmitterIterator(
```

Comment:
Let's remove the transient error trace emitters.

Context : https://github.com/airbytehq/airbyte/pull/42421

Reply:
i see, removed.